lib/commons/dom/get-rect-stack.js

```diff
@@ -475,21 +477,34 @@ export function getRectStack(grid, rect, recursed = false) {
   // went with pixel perfect collision rather than rounding
   const row = (y / gridSize) | 0;
   const col = (x / gridSize) | 0;
-  let stack = grid.cells[row][col].filter(gridCellNode => {
-    return gridCellNode.clientRects.find(clientRect => {
-      const rectX = clientRect.left;
-      const rectY = clientRect.top;
-
-      // perform an AABB (axis-aligned bounding box) collision check for the
-      // point inside the rect
-      return (
-        x <= rectX + clientRect.width &&
-        x >= rectX &&
-        y <= rectY + clientRect.height &&
-        y >= rectY
-      );
-    });
-  });
+
+  // we're making an assumption that there cannot be an element in the
+  // grid which escapes the grid bounds. For example, if the grid is 4x4 there
+  // can't be an element whose midpoint is at column 5. If this happens this
+  // means there's an error in our grid logic that needs to be fixed
+  if (row > grid.cells.length || col > grid._numCols) {
```

Comment:
I don't think we should be checking private variables here. They're private for a reason.